src/main/java/appeng/tile/AEBaseTile.java

```diff
@@ -280,52 +281,36 @@ public NBTTagCompound getUpdateTag()
 	public void handleUpdateTag( NBTTagCompound tag )
 	{
 		final ByteBuf stream = Unpooled.copiedBuffer( tag.getByteArray( "X" ) );
-		if( this.readFromStream( stream ) )
+
+		if( this.readUpdateData( stream ) )
 		{
 			this.markForUpdate();
 		}
 	}
 
-	private final boolean readFromStream( final ByteBuf data )
+	protected boolean readFromStream( final ByteBuf data ) throws IOException
 	{
-		boolean output = false;
-
-		try
+		if( this.canBeRotated() )
 		{
+			final EnumFacing old_Forward = this.forward;
+			final EnumFacing old_Up = this.up;
 
-			if( this.canBeRotated() )
-			{
-				final EnumFacing old_Forward = this.forward;
-				final EnumFacing old_Up = this.up;
-
-				final byte orientation = data.readByte();
-				this.forward = EnumFacing.VALUES[orientation & 0x7];
-				this.up = EnumFacing.VALUES[orientation >> 3];
-
-				output = this.forward != old_Forward || this.up != old_Up;
-			}
+			final byte orientation = data.readByte();
+			this.forward = EnumFacing.VALUES[orientation & 0x7];
+			this.up = EnumFacing.VALUES[orientation >> 3];
 
-			this.renderFragment = 100;
-			for( final AETileEventHandler h : this.getHandlerListFor( TileEventType.NETWORK_READ ) )
-			{
-				if( h.readFromStream( this, data ) )
-				{
-					output = true;
-				}
-			}
-
-			if( ( this.renderFragment & 1 ) == 1 )
-			{
-				output = true;
-			}
-			this.renderFragment = 0;
+			return this.forward != old_Forward || this.up != old_Up;
 		}
-		catch( final Throwable t )
+		return false;
+	}
+
+	protected void writeToStream( final ByteBuf data ) throws IOException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)